### PR TITLE
feat: add option to add description to tags

### PR DESCRIPTION
**Pull request by same-id**
#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md)?

Yes

#### Brief description of what is fixed or changed

Adding option to add description to tags by allowing specifying openapiv2 tags

### CODE REVIEW

**Review by johanbrandhorst — COMMENTED** [review]
Hi, thanks for your PR! This looks pretty neat to me, almost deceptively so. I'm going to rope in @ivucica for his thoughts on the changes since he's the original author.

**Comment by johanbrandhorst** [comment]
Looks like we ran into some CI errors

**Comment by same-id** [comment]
Hmm, seems like lint failed on removal of reserved fields, that is the correct process for that? Isn't it supposed to fail?

File: protoc-gen-openapiv2/options/openapiv2.proto [hunk 1]
```diff
@@ -92,10 +92,9 @@ message Swagger {
   // (that is, there is a logical OR between the security requirements).
   // Individual operations can override this definition.
   repeated SecurityRequirement security = 12;
-  // field 13 is reserved for 'tags', which are supposed to be exposed as and
-  // customizable as proto services. TODO(ivucica): add processing of proto
-  // service objects into OpenAPI v2 Tag objects.
-  reserved 13;
+  /// A list of tags for API documentation control. Tags can be used for logical
```
Comment (johanbrandhorst) [inline protoc-gen-openapiv2/options/openapiv2.proto:95]:
```suggestion
  // A list of tags for API documentation control. Tags can be used for logical
```

**Review by johanbrandhorst — COMMENTED** [review]
I would ignore that lint error, we can merge over it. Your change looks right to me. The error here assumes that the field number was reserved because it has been removed (https://docs.buf.build/breaking/rules#reserved_message_no_delete), which is not the case here.

**Comment by same-id** [comment]
Cool, committed the suggestion as well

**Comment by kkolur** [comment]
Thanks @same-id for this great work! 

@johanbrandhorst can we add [extensions](https://swagger.io/docs/specification/openapi-extensions/) to the Tag object as well, maybe in a separate PR?

**Comment by johanbrandhorst** [comment]
> @johanbrandhorst can we add [extensions](https://swagger.io/docs/specification/openapi-extensions/) to the Tag object as well, maybe in a separate PR?

I think this'd have to start as a separate issue since it's pretty broad scope, could you create one?

**Comment by kkolur** [comment]
> > @johanbrandhorst can we add [extensions](https://swagger.io/docs/specification/openapi-extensions/) to the Tag object as well, maybe in a separate PR?
> 
> I think this'd have to start as a separate issue since it's pretty broad scope, could you create one?

Created issue [here](https://github.com/grpc-ecosystem/grpc-gateway/issues/2942)

**Comment by kkolur** [comment]
@same-id @johanbrandhorst what would happen in the case a user defines their own tags at root level of open api object but dont disable the default rendering of service tags? Would the user defined tags override the default service tag generation, or does it get appended to it? How should this work?

**Comment by johanbrandhorst** [comment]
> @same-id @johanbrandhorst what would happen in the case a user defines their own tags at root level of open api object but dont disable the default rendering of service tags? Would the user defined tags override the default service tag generation, or does it get appended to it? How should this work?

The example added here already shows what happens - the tags are simply appended to the end of the service tags.

**Comment by johanbrandhorst** [comment]
Please rebase/merge on `master` and regenerate.

**Comment by same-id** [comment]
After I regenerate I'm having lots of unrelated `swagger-codegen` diffs, not sure why:

```
-                  "$ref": "#/definitions/examplepbNumericEnum"
+                  "$ref": "#/definitions/examplepbNumericEnum",
+                  "description": "Numeric enum description.",
+                  "title": "Numeric enum title"
                 },
                 "repeatedStringAnnotation": {
                   "type": "array",
@@ -2358,7 +2360,9 @@
                   "title": "Repeated nested object title"
                 },
                 "nestedAnnotation": {
-                  "$ref": "#/definitions/ABitOfEverythingNested"
+                  "$ref": "#/definitions/ABitOfEverythingNested",
+                  "description": "Nested object description.",
+                  "title": "Nested object title"
                 },
```

Can you try and execute swagger-codegen (version 2.4.8) on master and check if it's not in sync?

**Comment by johanbrandhorst** [comment]
You need to reinstall the plugin again, there was a change in the latest PR. Run `make install` before running the regeneration command.

**Review by johanbrandhorst — APPROVED** [review]
LGTM, and I chatted with @ivucica about this change, he's happy too.

**Comment by same-id** [comment]
Rebased

**Comment by kkolur** [comment]
@same-id can we get this over the hump? I have an issue that depends on changes here and would love to get started on that. Great work btw!

**Comment by johanbrandhorst** [comment]
This failure is an expected false positive, I will merge as-is.

**Comment by johanbrandhorst** [comment]
Thanks for your contribution @same-id!

**Comment by same-id** [comment]
Is there a community for grpc-gateway?
I would love to ask some general questions (mainly regarding integrations with redoc) and not sure where it's best to have that discussion.

**Comment by adambabik** [comment]
@same-id join [Gophers Slack](https://gophers.slack.com) and find the `grpc-gateway` channel. I think that's the most active community currently.

**Pull request merged**